File: Updog.Application/User/Common/UserLogin.cs
using Updog.Domain;

namespace Updog.Application {
    /// <summary>
    /// User login record and the issued auth token.
    /// </summary>
    public sealed class UserLogin {
        #region Properties
        /// <summary>
        /// The user it belongs to.
        /// </summary>
        public UserView User { get; }

        /// <summary>
        /// The auth token they were issued.
        /// </summary>
        /// <value></value>
        public string AuthToken { get; }
        #endregion

        #region Constructor(s)
        /// <summary>
        /// Create  new user login.
        /// </summary>
        /// <param name="user">The user of the login.</param>
        /// <param name="authToken">Their JWT.</param>
        public UserLogin(UserView user, string authToken) {
            User = user;
            AuthToken = authToken;
        }
        #endregion
    }
}

File: Updog.Client/src/user/user-login.ts
import { User } from './user';

/**
 * Login information sent back by the backend.
 */
export class UserLogin {
    /**
     * The user the login belongs to.
     */
    public user: User;

    /**
     * The auth token issued by the back end.
     */
    public authToken: string;

    /**
     * Create a new user login.
     * @param user The user of the login.
     * @param authToken Their auth token (JWT).
     */
    constructor(user: User, authToken: string) {
        this.user = user;
        this.authToken = authToken;
    }
}
